### ISSUE TITLE
support enough primitives to implement exists and for_all in parallel

### ISSUE THREAD

**Issue by craff** [ocaml-multicore/domainslib#89]
This typically requires propagating exception and cancellation.

**Comment by gasche** [comment]
In #90 I propose

```ocaml
val parallel_find : ?chunk_size:int -> start:int -> finish:int ->
  body:(int -> 'a option) -> pool -> 'a option
```

which lets one easily derive `exists`, `forall` but also value-returning versions.

**Comment by gasche** [comment]
I believe that this can be closed now that #90 is merged (thanks!).
